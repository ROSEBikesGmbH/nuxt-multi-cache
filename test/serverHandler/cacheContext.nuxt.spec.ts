import { beforeEach, beforeAll, describe, expect, test, vi } from 'vitest'
import cacheContextHandler from './../../src/runtime/serverHandler/cacheContext'
import { mockNuxtImport } from '@nuxt/test-utils/runtime'

vi.mock('#multi-cache-server-options', () => {
  return {
    default: {
      enabledForRequest: (event: any) => {
        return Promise.resolve(!!event.__ENABLED_FOR_REQUEST)
      },
    },
  }
})

mockNuxtImport('useRuntimeConfig', () => {
  return () => {
    return {
      multiCache: {
        component: true,
      },
    }
  }
})

describe('cacheContext server handler', () => {
  beforeEach(() => {
    vi.resetAllMocks()
    vi.resetModules()
  })
  beforeAll(() => {
    vi.resetModules()
  })
  test('Adds cache context to event', async () => {
    const event: any = {
      context: {},
      __ENABLED_FOR_REQUEST: true,
    }
    await cacheContextHandler(event)
    expect(event.context).toMatchSnapshot()
  })

  test('Does not add cache context to event if enabledForRequest returns false', async () => {
    const event: any = {
      context: {},
      __ENABLED_FOR_REQUEST: false,
    }
    await cacheContextHandler(event)
    expect(event.context).toEqual({})
  })
})
