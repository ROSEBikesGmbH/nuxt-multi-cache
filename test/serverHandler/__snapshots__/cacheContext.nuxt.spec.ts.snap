// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cacheContext server handler > Adds cache context to event 1`] = `
{
  "__MULTI_CACHE": {
    "component": {
      "clear": [Function],
      "dispose": [Function],
      "getItem": [Function],
      "getItemRaw": [Function],
      "getItems": [Function],
      "getKeys": [Function],
      "getMeta": [Function],
      "getMount": [Function],
      "getMounts": [Function],
      "hasItem": [Function],
      "mount": [Function],
      "removeItem": [Function],
      "removeMeta": [Function],
      "setItem": [Function],
      "setItemRaw": [Function],
      "setItems": [Function],
      "setMeta": [Function],
      "unmount": [Function],
      "unwatch": [Function],
      "watch": [Function],
    },
  },
  "__MULTI_CACHE_ROUTE": NuxtMultiCacheRouteCacheHelper {
    "cacheable": null,
    "maxAge": null,
    "tags": [],
  },
}
`;
